src/QMCDrivers/WFOpt/QMCFixedSampleLinearOptimizeBatched.cpp

```diff
@@ -238,6 +241,11 @@ void QMCFixedSampleLinearOptimizeBatched::engine_start(cqmc::engine::LMYEngine<V
 {
   app_log() << "entering engine_start function" << std::endl;
 
+  engineData forCostFunc;
```

Comment:
engineData is a misnamed member, please see [Naming](https://qmcpack.readthedocs.io/en/develop/developing.html#naming). Why is it initialized at engine start, and from here on I guess we won't know whether a method might require engine access.

I Don't like this pattern.  If QMCFixedSampleLinearOptimizeBatched composes the engines they should be present and valid as early in the life cycle as possible and the way they are composed into the object should give a clue to ownership and their lifetime.